text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+
+- [Hackage](https://hackage.haskell.org) (Haskell) has both keywords and
+uncurated categories without any metadata.
+
+- [NPM](https://www.npmjs.com) (JavaScript) has keywords.
+
+- [pub.dev](https://pub.dev) (Dart) has keywords under the name "topics".
+
+- [Maven Central](https://mvnrepository.com/repos/central) (Java),
+[NuGet](https://www.nuget.org) (.NET), [Packagist](https://packagist.org)
+(PHP) have keywords under the name "tags".
+
+- [Dub](https://code.dlang.org) (D), [Go](https://pkg.go.dev),
+[RubyGems](https://rubygems.org) (Ruby) have no keywords or categories.
+
+
+# Proposal / Motivation
+
+This RFC proposes to migrate crates.io from using categories **and** keywords
+to using only keywords, with a list of recommended keywords to address
+the duplication and clarity concerns from above. Keywords will keep their
+structure as a flat hierarchy, allowing crate owners to express nested
+categories can keyword combinations instead.
+
+An automated migration from categories to keywords for existing crates is
+currently not planned as the majority of crates that use categories also use
+keywords at the same time, while the reverse is not the case (see "Statistics"
+section above).
+
+The lists of pros and cons above already hint at a couple of reasons that
+motivate this RFC:
+
+The first reason being the confusion and complexity that is caused by having two
+competing concepts implemented right next to each other. This makes it hard for
+Rust users to figure out which concept to use when browsing and searching on
+crates.io. It also means extra work for crate authors that currently have to
+choose both keywords *and* categories for their crates.
+
+As can be seen in the "Prior art" section above, most package registries stick
+to either keywords or categories, and only the Haskell ecosystem is also
+implementing both.
+
+Another reason is the friction that is caused by the crates.io team having to
+decide upfront on whether to introduce a new category or not. As written above,
+this is seen by some people as legitimizing certain categories of crates and
+thus being a political decision. If the "recommended keywords" concept is
+implemented primarily based on usage data of existing keywords, this discussion
+can largely be sidestepped, and it gives the Rust community the power to decide
+what keywords should be recommended and presented on the crates.io frontpage.
+
+Finally, having the community decide on the keywords recommendations through
+their usage data reduces the amount of bike-shedding discussions that currently
+exist when categories are decided upon with very little up-front usage data.
+
+
+# Drawbacks
+
+- As mentioned above, the chance for having duplicates is increased since not
+everyone will use the list of recommended keywords. This can be countered to
+some degree with aliases for recommended keywords, but can't be prevented
+completely.
```

Comment:
Worth mentioning that this is *also* a drawback of the existing system, since many authors include keywords that are similar to categories instead, and there's duplication there as well.

Reply:
that's very true. see https://github.com/rust-lang/crates.io/issues/7306 as an example

any phrasing suggestions?

Reply:
How about this?

```suggestion
- As mentioned above, the chance for having duplicates is increased since not
everyone will use the list of recommended keywords. This can be countered to
some degree with aliases for recommended keywords, but can't be prevented
completely. That said, we already have duplicates between categories and
keywords today, so, moving to just keywords is a strict improvement.
```

Reply:
"strict improvement" sounds a bit absolute to me. can we soften this up a bit?

Reply:
@clarfonthey any thoughts on this? looks like it's the last blocker for merging this :)